### DISCUSSION TITLE
Arbitrary or custom spacing size in drawText (justify, center or right alignment)

### DISCUSSION THREAD

**Discussion by Canzervero** [Hopding/pdf-lib#1197] [Q&A]
Hi!

I'm using pdf-lib to create pdf files and is awesome, however, drawText don't has option to justify, center or right alignment, therefore, I have developed functions to make it possible, here evidences of results: [justified](https://drive.google.com/file/d/15UBa-L5U9gCNCIxk4kOOT9ciDpeMaEWi/view?usp=sharing), [centered](https://drive.google.com/file/d/1uLE9Otg5ztxR9kpZKiwV0bnR4eJPE_6k/view?usp=sharing) and [right](https://drive.google.com/file/d/1kfIe3d_7s66YhTjLMT9sip8UrH1qNDFf/view?usp=sharing) alignement.

The solution is not perfect because as could be seen, the alignement to right margin in the justified and right files, there is a little spacing (milimetric) that not is filled in.

For the above, someone known how to insert arbitrary o custom spacing size in the text when drawText function is used? Is that possible?

Thanks!

**Comment by xgg94** [comment]
Hi! Sorry, I can't help you with your question .. But what you build is exactly what I would need .. Is there a possibility for sharing your code? It would be so nice to add justification ...
Thank you
